### ISSUE TITLE
width/height `stretch` prefixes missing `-webkit-` version

### ISSUE THREAD

**Issue by lunelson** [postcss/autoprefixer#1294]
According to [caniuse](https://caniuse.com/#feat=mdn-css_properties_width_stretch), Safari should get a `-webkit-stretch` version of `stretch` width/height rule

![image](https://user-images.githubusercontent.com/1242864/77299221-6c1ad180-6cec-11ea-9e3e-dc34bb57919f.png)

...however this doesn't appear to be the case with the current demo at https://autoprefixer.github.io/, using the "last 10 years" config:

### Input

```css
/* pre-prefixed */
.stretch {
  width: -webkit-fill-available;
  width: -webkit-stretch;
  width: -moz-available;
  width: stretch;
}

/* non-prefixed */
.stretch {
  width: stretch;
}
```

### Output
```css
/* pre-prefixed */
.stretch {
  width: -webkit-fill-available;
  width: -webkit-stretch;
  width: -moz-available;
  width: stretch;
}

/* non-prefixed */
.stretch {
  width: -webkit-fill-available;
  width: -moz-available;
  width: stretch;
}
```

**Comment by ai** [comment]
Thanks for good issue with all details.

Fixed a38e426

**Comment by ai** [comment]
Released in 9.7.5

**Comment by ai** [comment]
BTW, I like the design of [your website](https://lunelson.xyz/)

**Comment by lunelson** [comment]
Thanks @ai that's so nice to hear! And thanks for the quick fix 😉 !

**Comment by simevidas** [comment]
The `width: stretch` value seems to be non-standard and, more importantly, unspecified. Maybe Autoprefixer should not support such features until they are ready for use.

@lunelson Could you describe your use-case? I’m curious what this value does and what it is useful for.

**Comment by ai** [comment]
Hm, I thought that I saw it in some previous draft.

Maybe they changed the draft because I am still founding [it’s mentions](https://drafts.csswg.org/css-sizing-4/#block-intrinsic) but without definitions.

**Comment by simevidas** [comment]
It was removed from Level 3 in 2017: https://github.com/w3c/csswg-drafts/commit/f5f655bc98a8dfce25473ecf563f230fa3bf6d35.

It will probably be added to Level 4 at some point, but right now there is no definition.

**Comment by lunelson** [comment]
@simevidas my use-case had to do with ensuring that block items with negative horizontal margins (`left`, `right` or both) would expand to the full possible width (which strangely does not happen in all browsers under all conditions). Previously I had done it with explicit widths (`.foo`), but `stretch` does not require knowing the other values (`.bar`):

```css
.foo, .bar {
  margin: 0 -2rem;
}
.foo {
  width: calc(100% + 4rem);
}
.bar {
  width: stretch;
}
```
> The width: stretch value seems to be...completely unspecified. Maybe Autoprefixer should not support such features until they are ready to use.

I'll defer to you and @ai on supporting unspecified features; but this one *was* specified and then removed, thus at least in this case "unspecified" is not strictly the same as "ready to use". This feature works in the majority of browsers, if prefixed, and I have used it successfully.

**Comment by simevidas** [comment]
@tomayac informed me that `-webkit-stretch` does not exist. It’s not supported in the desktop version of Safari (see screenshot). The linked Can I Use table is based on MDN data. I’m assuming that that data is incorrect. It wouldn’t be the first time MDN is incorrect.

So yeah, this may need to be reverted. 😅

<img width="448" alt="Screen Shot 2020-04-06 at 12 56 18 AM" src="https://user-images.githubusercontent.com/716405/78512026-800d0b80-77a1-11ea-8da6-81dc7089f1c6.png">

**Comment by tomayac** [comment]
Tested in Safari Technology Preview (`Release 103 (Safari 13.2, WebKit 15610.1.7.3.2)`).

**Comment by lunelson** [comment]
...so this shows that the `caniuse` data is out of date, but are we sure that _no version of Safari_ has ever used the `-webkit-stretch` prefix?

**Comment by tomayac** [comment]
> ...so this shows that the caniuse data is out of date

https://github.com/mdn/browser-compat-data/pull/5955

> are we sure that no version of Safari has ever used the -webkit-stretch prefix?

I checked the history of https://github.com/WebKit/webkit/blob/master/Source/WebCore/css/CSSValueKeywords.in to no avail. It didn't exist as far as I can tell.

**Comment by simevidas** [comment]
@lunelson Not out of date but incorrect. The `-webkit-fill-available` value predates Chrome’s engine fork. When this value was implemented, Chrome was still using WebKit, Safari’s current engine.

Also, the `stretch` value was only introduced in 2017 (https://github.com/w3c/csswg-drafts/commit/c372e95e1438308623941bb28ef0f5e7e31dffb3). If Apple added `-webkit-stretch` as an alias for `-webkit-fill-available`, they would have to have done it fairly recently.

**Comment by ai** [comment]
I will revert the commit today. Thanks for the very deep investigating.

**Comment by ai** [comment]
@lunelson don't worry, you were right of creating the issue. It helped us find the problem in MDN docs.

**Comment by ai** [comment]
Revert was released at 9.7.6

**Comment by lunelson** [comment]
Thanks @ai, no worries, I'm glad the deeper discovery was made. I just hope this feature remains in, since as I said, these are working shipped features 🙏!
